hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableConfig.java

```diff
@@ -737,11 +738,12 @@ public boolean isMetadataTableAvailable() {
   /**
    * Checks if metadata table is enabled and the specified partition has been initialized.
    *
-   * @param partition The partition to check
+   * @param metadataPartitionType The metadata table partition type to check
    * @returns true if the specific partition has been initialized, else returns false.
    */
-  public boolean isMetadataPartitionAvailable(MetadataPartitionType partition) {
-    return getMetadataPartitions().contains(partition.getPartitionPath());
+  public boolean isMetadataPartitionAvailable(MetadataPartitionType metadataPartitionType) {
```

Comment:
Is this sufficient or should we actually check the whole partition-path. Something like `public boolean isMetadataPartitionPathAvailable(String partitionPath)`. Mostly asking for the case when there could be multiple functional-index.

Reply:
i changed the logic a bit. We will still find func index in the mdt partitions to initialize, but then when we actually go about initializing we will check based on full partition path - see `initializeFromFileSystem` and `getFunctionalIndexPartitionsToInit` in `HoodieBackedTableMetadataWriter`

Reply:
Loosk good. The test failures look like simple checkstyle fixes.